pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go

```diff
@@ -140,57 +142,53 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{RequeueAfter: refreshInt}, nil
 }
 
-func (r *Reconciler) resolveExternalSecret(ctx context.Context, clusterExternalSecret *esv1beta1.ClusterExternalSecret, existingES *metav1.PartialObjectMetadata, namespace v1.Namespace, esName string, esMetadata esv1beta1.ExternalSecretMetadata) (string, error) {
-	// this means the existing ES does not belong to us
-	if err := controllerutil.SetControllerReference(clusterExternalSecret, existingES, r.Scheme); err != nil {
```

Comment:
Using `controllerutil.SetControllerReference` is kind of clever but the problem is there are actually two places where check owner reference, in `resolveExternalSecret` method and `checkForError` function. I think this makes it hard to understand how we validate the owner reference, so I extracted the logic to one function, `isExternalSecretOwnedBy`.